### PR TITLE
Use PHP 7.1 syntax

### PR DESCRIPTION
**Pull request by keradus**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | n/a
| Fixed tickets | #22733
| License       | MIT
| Doc PR        | n/a

After #22733, it's possible to use new language syntax

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
This is going to create a hell lot of merge conflicts.
All changes that are pure CS should be reverted IMHO.
And the remaining ones challenged for technical benefit.

**Comment by keradus** [comment]
That's why I kept different changes in different commits.
During merging lower to higher branch one could accept lower-branch version of conflicted code and re-apply styles.
On the other hand, probably there is not much value in keeping PRs open even since 2012.
While bumping reqs to 7.1 is great, keeping syntax from 5.3 is not...

**Comment by theofidry** [comment]
Now that constants can be private maybe some should be reviewed to see if they should be private and not public? As making them private would be too big of a BC, we could go with the `@private` tag on them.

**Comment by keradus** [comment]
I agree with that @theofidry . Without accessor, it's public, so let write it instead of letting the compiler assume that. If some of them could not be public, it shall be changed in separated PRs with proper BC breaking changelog

File: src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterEventListenersAndSubscribersPass.php [hunk 1]
```diff
@@ -65,8 +65,8 @@ public function process(ContainerBuilder $container)
         $this->container = $container;
         $this->connections = $container->getParameter($this->connections);
         $sortFunc = function ($a, $b) {
-            $a = isset($a['priority']) ? $a['priority'] : 0;
-            $b = isset($b['priority']) ? $b['priority'] : 0;
+            $a = $a['priority'] ?? 0;
+            $b = $b['priority'] ?? 0;
 
             return $a > $b ? -1 : 1;
```
Comment (Taluu) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterEventListenersAndSubscribersPass.php]: retun `$a <=> $b1` ?

**Comment by nicolas-grekas** [comment]
:-1: for sure and for the same reason we keep using `array()` vs `[]`: this is pure code style change. It will create merge conflicts that we'll have to deal with for years to come.

Note that @theofidry's idea LGTM: replacing `@internal` consts by `private const` when possible.

**Comment by javiereguiluz** [comment]
@keradus I wish we could use these new features, but we can't. To avoid merging issues, we can only use PHP 7.1 for new features added to master.

For the existing code, we can only upgrade when the last Symfony version using the old PHP version finishes its support. This means that:

* We could upgrade Symfony code to use PHP 5.5 on November 2019 (when Symfony 2.8 dies)
* We could upgrade Symfony code to use PHP 7.1 on November 2021 (when Symfony 3.4 dies).

**Comment by keradus** [comment]
that sadly means code would always follow few-years old syntax :(
I bet there is a way to avoid merging hell, probably first step would be to close outdated PRs...

**Comment by nicolas-grekas** [comment]
Infact, merging hell is not related to old PRs. It's related to the fact that several times per week, we merge 2.7 into 2.8 into 3.2 into 3.3 into 3.4 into master. Any change in any branch <=3.4 will way more likelly trigger a merge conflict when 3.4 is then merged into master.

**Comment by keradus** [comment]
what about accepting version of lower branch and reapply CS as conflict fix ?

**Comment by stof** [comment]
@keradus this won't work, because it may break newer features if the conflict is not *strictly* about coding standards

**Comment by nicolas-grekas** [comment]
Closing as explained, that's our burden :)

File: src/Symfony/Bridge/PhpUnit/Legacy/TestRunner.php [hunk 2]
```diff
@@ -27,7 +27,7 @@ protected function handleConfiguration(array &$arguments)
 
         $result = parent::handleConfiguration($arguments);
 
-        $arguments['listeners'] = isset($arguments['listeners']) ? $arguments['listeners'] : array();
+        $arguments['listeners'] = $arguments['listeners'] ?? array();
```
Comment (SofHad) [inline src/Symfony/Bridge/PhpUnit/Legacy/TestRunner.php]:
```php
`$arguments['listeners'] = $arguments['listeners'] ?? [];
```

**Comment by keradus** [comment]
in that case milestone could be misleading

**Pull request closed without merging**